build-system/tasks/runtime-test/unit.js

```diff
@@ -0,0 +1,236 @@
+  if (argv.nobuild) {
+    return;
+  }
+
+  if (argv.local_changes) {
+    return css();
+  }
+
+  await clean();
+  await dist();
+}
+
+function setConfigOverrides(config) {
+  config.singleRun = !argv.watch && !argv.w;
+  config.client.mocha.grep = !!argv.grep;
+  config.client.verboseLogging = !!argv.verbose || !!argv.v;
+  config.client.captureConsole =
+    !!argv.verbose || !!argv.v || !!argv.local_changes;
+
+  config.browserify.configure = function(bundle) {
+    bundle.on('prebundle', function() {
+      log(green('Transforming tests with'), cyan('browserify') + green('...'));
+    });
+    bundle.on('transform', function(tr) {
+      if (tr instanceof babelify) {
+        tr.once('babelify', function() {
+          process.stdout.write('.');
+        });
+      }
+    });
+  };
+
+  // config.client is available in test browser via window.parent.karma.config
+  config.client.amp = {
+    useCompiledJs: false, // never compiled for unit tests
+    saucelabs: !!argv.saucelabs_lite,
+    singlePass: false, // never single pass for unit tests
+    adTypes: getAdTypes(),
+    mochaTimeout: defaultConfig.client.mocha.timeout,
+    propertiesObfuscated: false, // never single pass for unit tests
+    testServerPort: defaultConfig.client.testServerPort,
+    testOnIe: config.browsers.includes('IE'),
+  };
+}
+
+function getFileConfig() {
+  const files = testConfig.commonUnitTestPaths.concat(
+    testConfig.chaiAsPromised
+  );
+
+  if (argv.saucelabs_lite) {
+    return {'files': files.concat(testConfig.unitTestOnSaucePaths)};
+  }
+
+  if (argv.local_changes) {
+    return {'files': files.concat(getUnitTestsToRun(testConfig.unitTestPaths))};
+  }
+
+  return {'files': files.concat(testConfig.unitTestPaths)};
+}
+
+function getReporterConfig() {
+  let {reporters} = defaultConfig;
+
+  if (
+    (!isTravisBuild() && (argv.testnames || argv.local_changes)) ||
+    (argv.files && !argv.saucelabs_lite)
+  ) {
+    reporters = ['mocha'];
+  }
+
+  if (argv.coverage) {
+    reporters.push('coverage-istanbul');
+  }
+
+  if (argv.saucelabs_lite) {
+    reporters.push('saucelabs');
+  }
+
+  return {'reporters': reporters};
+}
+
+function getTestConfig() {
+  const browsers = getBrowserConfig();
+  const files = getFileConfig();
+  const reporters = getReporterConfig();
+  const config = Object.assign({}, defaultConfig, browsers, files, reporters);
+
+  maybeSetCoverageConfig(config, 'lcov-unit.info');
+  setConfigOverrides(config);
+
+  return config;
+}
+
+function setup(config) {
+  // Avoid Karma startup errors
+  refreshKarmaWdCache();
+
+  const testServer = startTestServer(config.client.testServerPort);
+  const handlerProcess = createCtrlcHandler('gulp unit');
+
+  return new Map()
+    .set('handlerProcess', handlerProcess)
+    .set('testServer', testServer);
+}
+
+function teardown(env) {
+  const exitCode = env.get('exitCode');
+
+  // Exit tests
+  // TODO(rsimha, 14814): Remove after Karma / Sauce ticket is resolved.
```

Comment:
I think this is resolved. Let's try and remove the `setTimeout` logic below in a separate PR.